SDWebImage/Core/SDWebImageDownloaderRequestModifier.m

```diff
@@ -53,12 +53,9 @@ - (instancetype)initWithBody:(NSData *)body {
 }
 
 - (instancetype)initWithMethod:(NSString *)method headers:(NSDictionary<NSString *,NSString *> *)headers body:(NSData *)body {
-    method = [method copy];
-    headers = [headers copy];
-    body = [body copy];
     return [self initWithBlock:^NSURLRequest * _Nullable(NSURLRequest * _Nonnull request) {
```

Comment:
Emm. I'm a little confused. Why doesn't this place need to be `copy`? 

I'm not sure `[request copy]` cause a    HTTPBody&HTTPMethod&headers  `copy`?

Reply:
No usage.

Reply:
See NSHTTPRequest.HTTPMethod, it's already marked `@property (copy)`

Reply:
This change, even the input args is a `NSMutableString` or `NSMutableData`, can move the actual copy step, from `init` method, into the `modifedRequestWithRequest:` protocol method.

This may help for some really small performance issue. But why not.

Reply:
Yes, You are right!

Reply:
To avoid cases that user pass the NSMutableDictionary and then call `addObject:` and `modifiedRequestWithRequest:` in different thread, I still keep the copy code...